src/components/implementation/no_interface/new_boot/boot_deps.h

```diff
@@ -28,24 +38,24 @@ boot_comp_map_memory(struct cobj_header *h, spdid_t spdid, pgtblcap_t pt)
 	int flag;
 	vaddr_t dest_daddr, prev_map = 0;
 	int tot = 0, n_pte = 1;	
-	
 	struct cobj_sect *sect = cobj_sect_get(h, 0);
 	
-	/*Expand Page table, could do this faster*/
-	for(j = 0; j < (int)h->nsect; j++){
+	/* Expand Page table, could do this faster */
```

Comment:
See the new attempt, this function is back in booter.c. the old booter has the whole function in the booter.c. I added to small functions to encapsulate the mechanics that may be not generic. is this what you're thinking of? I now realize the naming of the boot_comp_mmap() function might be pretty stupid. I can fix that if you think this is appropriate.